File: fx/src/main/java/gargoyle/fx/FXUserException.java
package gargoyle.fx;

import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;
import java.io.Serial;
import java.util.Map;

public final class FXUserException extends FXException {
    @Serial
    private static final long serialVersionUID = -1;

    public FXUserException(final String code, final Map<String, ?> args) {
        super(FXUtil.format(code, args));
    }

    public FXUserException(final Throwable cause, final String code, final Map<String, ?> args) {
        super(FXUtil.format(code, args), cause);
    }

    @Serial
    private void readObject(final ObjectInputStream s) {
        throw new IllegalStateException(FXUserException.class.getName());
    }

    @Serial
    private void writeObject(final ObjectOutputStream s) {
        throw new IllegalStateException(FXUserException.class.getName());
    }
}

File: fx/src/main/kotlin/gargoyle/fx/FxUserException.kt
package gargoyle.fx

class FxUserException : FxException {
    constructor(code: String, args: Map<String, Any> = mapOf<String, String>()) :
            super(FxUtil.format(code, args))

    constructor(cause: Throwable?, code: String, args: Map<String, Any> = mapOf<String, String>()) :
            super(FxUtil.format(code, args), cause)

}
